abdiff/helpers/timdex_sources.py

```diff
@@ -0,0 +1,93 @@
+"""abdiff.helpers.timdex_sources"""
+
+import datetime
+import logging
+import re
+
+import boto3  # type: ignore[import-untyped]
+
+from abdiff.config import Config
+
+logger = logging.getLogger(__name__)
+
+CONFIG = Config()
+
+
+def get_extracted_files_for_source(
+    source: str,
+    bucket: str = CONFIG.TIMDEX_BUCKET,
+) -> list[str]:
+    """List S3 URIs for extract files in TIMDEX S3 bucket for a given source."""
+    s3_client = boto3.client("s3")
+    files = []
+
+    paginator = s3_client.get_paginator("list_objects_v2")
+    page_iterator = paginator.paginate(Bucket=bucket, Prefix=source)
+
+    for page in page_iterator:
+        if "Contents" in page:
+            for obj in page["Contents"]:
+                if not obj["Key"].endswith("/"):  # skip folders
+                    s3_uri = f"s3://{bucket}/{obj['Key']}"
+                    files.append(s3_uri)  # noqa: PERF401
+
+    # filter where "extracted" in filename
+    return [file for file in files if "extracted" in file]
+
+
+def get_ordered_extracted_files_since_last_full_run(source: str) -> list[str]:
+    """Get extract files, from last full run, through all daily runs, for a source."""
+    logger.info(f"Retrieving ordered extracted files for source: '{source}'")
+    all_files = get_extracted_files_for_source(source)
+
+    # Find all full extract files and extract their dates
+    full_extracts = [f for f in all_files if _is_full_extract(f)]
+    if not full_extracts:
+        logger.warning("No full extracts found.")
+        return []
+
+    # Extract dates from full extract files and find the most recent date
+    full_extract_dates = [_extract_date(f) for f in full_extracts]
+    most_recent_full_date = max(full_extract_dates)
+
+    # Collect all full extract files with the most recent date
+    most_recent_full_files = sorted(
+        f for f in full_extracts if _extract_date(f) == most_recent_full_date
+    )
```

Comment:
Question: `most_recent_full_files` is plural because of the possibility of sequenced files (i.e., files ending with an index like "01"), right? Is there ever a scenario where we'd have: two "full" extract files for the same source on the same date (not sequenced)? 🤔

Reply:
Exactly.  It used to be `most_recent_full_file`!